pkg/preparation/sqlrepo/schema.sql

```diff
@@ -0,0 +1,122 @@
+-- enable foreign key constraints
+PRAGMA foreign_keys = ON;
```

Comment:
https://stackoverflow.com/questions/64785806/why-is-sqlite-pragma-foreign-keys-not-persistent-across-db-connections worth a read, may need to set this on the connection level.

Feel free to crib from https://github.com/storacha/piri/tree/main/pkg/database/sqlitedb which attempts to provide a nice-ish interface for this.